profile_test.go

```diff
@@ -0,0 +1,313 @@
+package eat
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	InvalidProfile = 100
+	InvalidURL     = "abcd"
+	EmptyURL       = ""
+	InvalidOID     = "xxxx"
+	EmptyOID       = ""
+)
+
+// TestProfile_GetSet_Basic_URL tests the basic setting of Profile as URL string
+func TestProfile_GetSet_Basic_URL(t *testing.T) {
+	inputURL := "https://samplewebsite.com"
+
+	profile, err := NewProfile(inputURL)
+	assert.Nil(t, err)
+
+	expectedURL, err := profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedURL, inputURL)
+	assert.True(t, profile.IsURI())
+
+	inputURL = "https://samplenewwebsite.co.uk"
+	err = profile.Set(inputURL)
+	assert.Nil(t, err)
+
+	expectedURL, err = profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedURL, inputURL)
+	assert.True(t, profile.IsURI())
+
+	// Negative test cases, below
+	inputURL = InvalidURL
+	err = profile.Set(inputURL)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "profile decode failed no valid URL or OID")
+
+	inputURL = EmptyURL
+	err = profile.Set(inputURL)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "profile decode failed no valid URL or OID")
+}
+
+// TestProfile_GetSet_Basic_OID tests the saving and retrieval of OID as profile
+func TestProfile_GetSet_Basic_OID(t *testing.T) {
```

Comment:
this test too has a clear pattern and could be refactored into a table driven test